controllers/etcd_controller.go

```diff
@@ -159,30 +179,159 @@ func (r *EtcdReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
 		}
 	}
 	if err := r.addFinalizersToDependantSecrets(etcdCopy); err != nil {
-		if err := r.updateEtcdErrorStatus(etcdCopy, err); err != nil {
+		if err := r.updateEtcdErrorStatus(etcd, etcdCopy, err); err != nil {
 			return ctrl.Result{
 				Requeue:      true,
 				RequeueAfter: time.Second * 5,
-			}, nil
+			}, err
 		}
 	}
 
-	imageVector, err := imagevector.ReadGlobalImageVectorWithEnvOverride(imageYAMLPath)
-	if err != nil {
+	if err := r.reconcileEtcd(etcdCopy); err != nil {
+		if err := r.updateEtcdErrorStatus(etcd, etcdCopy, err); err != nil {
+			return ctrl.Result{
+				Requeue:      true,
+				RequeueAfter: time.Second * 5,
+			}, err
+		}
 		return ctrl.Result{
 			Requeue:      true,
 			RequeueAfter: time.Second * 5,
 		}, err
 	}
 
-	images, err := imagevector.FindImages(imageVector, []string{"etcd", "etcd-backup-restore"})
-	if err != nil {
+	if err := r.updateEtcdStatus(etcd); err != nil {
 		return ctrl.Result{
 			Requeue:      true,
 			RequeueAfter: time.Second * 5,
 		}, err
 	}
 
+	logger.Infof("Succesfully reconciled etcd: %s", etcd.GetName())
+
+	return ctrl.Result{
+		Requeue:      true,
+		RequeueAfter: time.Second * 5,
+	}, nil
+}
+
+func (r *EtcdReconciler) getEtcdService(etcd *druidv1alpha1.Etcd) (*corev1.ServiceList, error) {
+	etcdServiceList := &corev1.ServiceList{}
+	err := r.List(context.TODO(), etcdServiceList, client.InNamespace(etcd.Namespace), client.MatchingLabels(etcd.Spec.Labels))
+	return etcdServiceList, err
+}
+
+func (r *EtcdReconciler) getEtcdStatefulSet(etcd *druidv1alpha1.Etcd) (*appsv1.StatefulSetList, error) {
+	etcdStatefulSetList := &appsv1.StatefulSetList{}
+	err := r.List(context.TODO(), etcdStatefulSetList, client.InNamespace(etcd.Namespace), client.MatchingLabels(etcd.Spec.Labels))
+	return etcdStatefulSetList, err
+}
+
+func (r *EtcdReconciler) getEtcdConfigMap(etcd *druidv1alpha1.Etcd) (*corev1.ConfigMapList, error) {
+	etcdConfigMapList := &corev1.ConfigMapList{}
+	err := r.List(context.TODO(), etcdConfigMapList, client.InNamespace(etcd.Namespace), client.MatchingLabels(etcd.Spec.Labels))
+	return etcdConfigMapList, err
+}
+
+func (r *EtcdReconciler) getExistingEtcdResourcesAndSetOwnerRef(etcd *druidv1alpha1.Etcd) (bool, error) {
+	etcdServiceList, err := r.getEtcdService(etcd)
+	if err != nil {
+		return false, err
+	}
+
+	etcdStatefulsetList, err := r.getEtcdStatefulSet(etcd)
+	if err != nil {
+		return false, err
+	}
+
+	etcdConfigMapList, err := r.getEtcdConfigMap(etcd)
+	if err != nil {
+		return false, err
+	}
+
+	if (len(etcdServiceList.Items) + len(etcdStatefulsetList.Items) + len(etcdConfigMapList.Items)) == 0 {
+		return false, nil
```

Comment:
Can we log a message here which resources were found or not found?

Reply:
This logic not used anymore.